Modules/_cursesmodule.c

```diff
@@ -199,6 +222,33 @@ static char *screen_encoding = NULL;
                         "must call start_color() first");       \
         return 0; }
 
+#define CHECK_NOT_NULL_OR_ABORT(VALUE)  \
+    do {                                \
+        if ((VALUE) == NULL) {          \
+            goto abort;                 \
+        }                               \
+    } while (0)
+
+#define CHECK_RET_CODE_OR_ABORT(STATUS) \
+    do {                                \
+        if ((STATUS) < 0) {             \
+            goto abort;                 \
+        }                               \
+    } while (0)
+
+#define CHECK_RET_FLAG_OR_ABORT(STATUS) \
+    do {                                \
+        if (!(STATUS)) {                \
+            goto abort;                 \
+        }                               \
+    } while (0)
+
+#define ADD_INT_CONSTANT_OR_ABORT(MODULE, STRING, VALUE)            \
+    do {                                                            \
+        int rc = PyModule_AddIntConstant(MODULE, STRING, VALUE);    \
+        CHECK_RET_CODE_OR_ABORT(rc);                                \
+    } while (0)
```

Comment:
You should move these macros just before their first use, I don't think that it's useful to have them at the top.